content/blog/2022-02-25-react-v18-rc.md

```diff
@@ -0,0 +1,247 @@
+
+*Note for React Native users: React 18 will ship in React Native with the New React Native Architecture. For more information, see the [React Conf keynote here](https://www.youtube.com/watch?v=FZ0cG47msEk&t=1530s).*
+
+## Installing
+
+To install the latest React 18 RC, use the `@rc` tag:
+
+```bash
+npm install react@rc react-dom@rc
+```
+
+Or if you’re using yarn:
+
+```bash
+yarn add react@rc react-dom@rc
+```
+
+## New Client Root APIs
+
+When you first install React 18, you will see a warning in the console:
+
+![Screenshot of React error with text "Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17."](../images/blog/react-v18-rc/reactdom_warning.png)
+
+React 18 introduces a new root API which provides better ergonomics for managing roots. The new root API also enables the new concurrent renderer, which allows you to opt-into concurrent features.
+
+To upgrade `ReactDOM.render`:
+
+```js
+// Before
+const container = document.getElementById('app');
+ReactDOM.render(<App tab="home" />, container);
+
+// After
+const container = document.getElementById('app');
+const root = ReactDOM.createRoot(container);
+root.render(<App tab="home" />);
+
+```
+
+We’ve also changed `unmountComponentAtNode` to `root.unmount`:
+
+```js
+// Before
+unmountComponentAtNode(containter);
+
+// After
+root.unmount();
+```
+
+And removed the callback from `render`:
+
+```js
+// Before
+const container = document.getElementById('app');
+ReactDOM.render(<App tab="home" />, container, () => {
+  console.log('rendered');
+});
+
+// After
+function AppWithCallbackAfterRender() {
+  useEffect(() => {
+    console.log('rendered');
+  });
+
+  return <App tab="home" />
+}
+
+const container = document.getElementById('app');
+const root = ReactDOM.createRoot(container);
+root.render(<AppWithCallbackAfterRender />);
+```
+
+Finally, to upgrade `ReactDOM.hydrate`:
+
+```js
+// Before
+const container = document.getElementById('app');
+ReactDOM.hydrate(<App tab="home" />, container);
+
+// After
+const container = document.getElementById('app');
+const root = ReactDOM.hydrateRoot(container, <App tab="home" />);
+// Unlike with createRoot, you don't need a separate root.render() call here.
+```
+
+For more information, see the [working group discussion here](https://github.com/reactwg/react-18/discussions/5).
+
+## Updates to Server Rendering APIs
+
+In this release, we’re revamping our server rendering APIs to support Suspense on the server and Streaming SSR.
+
+As part of these changes, the following APIs will continue working, but with limited support for Suspense. Instead of erroring, these APIs will server render the nearest fallback for any subtree that suspends, and retry rendering on the client:
+
+* `renderToString`
+* `renderToStaticMarkup`
```

Comment:
renderToStaticMarkup won't "retry rendering on the client" because there's no client in this move, so I guess it just leaves it in permanent fallback.